filestore/fs.py

```diff
@@ -214,33 +252,136 @@ def get_spec_handler(self, resource):
 
         kwargs = resource['resource_kwargs']
         rpath = resource['resource_path']
+        root = resource.get('root', '')
+        rpath = os.path.join(root, rpath)
         ret = handler(rpath, **kwargs)
         h_cache[key] = ret
         return ret
 
 
 class FileStore(FileStoreRO):
-    def insert_resource(self, spec, resource_path, resource_kwargs):
+    def insert_resource(self, spec, resource_path, resource_kwargs, root=''):
```

Comment:
Maybe do some inspection here to:

1. Raise if the given path is not an absolute path.
2. If no root is given, strip off the leading '/' from the resource_path and make that the root.